_plugins/docs_generator.rb

```diff
@@ -64,6 +64,9 @@ def generate(site)
       documents = {}
       repo_build['documents'].each do |permalink, local_content|
         parent_path, * = permalink.rpartition('/')
+        while not parent_path.empty? and not documents.key? parent_path
```

Comment:
Does anything outside of this loop guarantee that parent pages are visited before child pages? Otherwise the parent may not be in `documents` yet?

Reply:
Yes, that's the undocumented post-condition of [`DocsPageGenerator::build_with_sphinx`](https://github.com/ros-infrastructure/rosindex/pull/195/files#diff-dac9c27c21ed6bc48485c0e6ed3dbb92R210-R233).

Reply:
Cool. I think a comment explaining as much would be helpful for people looking here in the future.

Reply:
Agreed. PTAL https://github.com/ros-infrastructure/rosindex/pull/195/commits/a95267f94a8ebd3e5b57c5801281661978a46806

Reply:
Looks good. Thanks.